src/daft-logical-plan/src/optimization/rules/reorder_joins/join_graph.rs

```diff
@@ -46,36 +51,76 @@ impl Display for JoinNode {
     }
 }
 
-/// JoinEdges currently represent a bidirectional edge between two relations that have
-/// an equi-join condition between each other.
-#[derive(Debug)]
-struct JoinEdge(JoinNode, JoinNode);
+#[derive(Clone, Debug)]
+pub(crate) struct JoinCondition {
```

Comment:
do these need to `pub(crate)` or just local to the file `pub(self)` or nothing at all